packages/mask/src/plugins/IdeaMarket/hooks/useFetchIdeaTokens.ts

```diff
@@ -0,0 +1,13 @@
+import { escapeRegExp } from 'lodash-unified'
+import { useAsync } from 'react-use'
+import { fetchAllTokens } from '../apis'
+
+export function useFetchIdeaTokensBySearch(searchText: string) {
+    const searchRegex = new RegExp(escapeRegExp(searchText), 'i')
+
+    const { value, error, loading } = useAsync(async () => {
+        return fetchAllTokens(searchText)
```

Comment:
Don't call API on the front page. Prefer to use a service RPC call.

Reply:
Did it but got en error that I do not understand, see picture below:
![Screenshot from 2022-03-24 15-51-30](https://user-images.githubusercontent.com/7540595/159920630-c643b9d0-4300-4f69-afdd-2385822bec37.png)

Do you have an idea what is wrong ?

Reply:
It looks not related to the RPC call.

Reply:
listings shows up without rpc call:
![image](https://user-images.githubusercontent.com/7540595/160442492-82b93784-33c9-4578-9e96-3bcea9c40872.png)


listings do not show up when rpc call, produces an infinite load:
![Screenshot from 2022-03-28 19-20-16](https://user-images.githubusercontent.com/7540595/160442873-c5d9b94a-53f6-4505-af54-0ad18ad46082.png)